build-system/server/typescript-compile.js

```diff
@@ -13,40 +13,55 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-const pathModule = require('path');
+const esbuild = require('esbuild');
+const globby = require('globby');
+const path = require('path');
 const {cyan, green} = require('kleur/colors');
+const {endBuildStep} = require('../tasks/helpers');
 const {exec} = require('../common/exec');
 const {log} = require('../common/logging');
 
-const SERVER_TRANSFORM_PATH = 'build-system/server/new-server/transforms';
-
-/**
- * @return {string}
- */
-function getBuildCmd() {
-  switch (process.platform) {
-    case 'win32':
-      return `node .\\node_modules\\typescript\\lib\\tsc.js -p ${SERVER_TRANSFORM_PATH.split(
-        '/'
-      ).join(pathModule.sep)}${pathModule.sep}tsconfig.json`;
-
-    default:
-      return `./node_modules/typescript/bin/tsc -p ${SERVER_TRANSFORM_PATH}/tsconfig.json`;
-  }
-}
+const SERVER_TRANSFORM_PATH = path.join(
+  'build-system',
+  'server',
+  'new-server',
+  'transforms'
+);
+const CONFIG_PATH = path.join(SERVER_TRANSFORM_PATH, 'tsconfig.json');
 
 /**
  * Builds the new server by converting typescript transforms to JS
- * @return {void}
+ * @return {Promise}
  */
-function buildNewServer() {
+async function buildNewServer() {
   log(
     green('Building'),
     cyan('AMP Server'),
     green('at'),
     cyan(`${SERVER_TRANSFORM_PATH}/dist`) + green('...')
   );
-  const result = exec(getBuildCmd(), {'stdio': ['inherit', 'inherit', 'pipe']});
+  const entryPoints = globby.sync(
+    path.join(SERVER_TRANSFORM_PATH, '**', '*.ts')
+  );
```

Comment:
If this doesn't work on Windows, this may need to be:
```suggestion
  const entryPoints = globby
    .sync(path.join(SERVER_TRANSFORM_PATH, '**', '*.ts'))
    .map(path.normalize);
```

I'm playing around with something similar for the Closure rewrite.